src/lib.rs

```diff
@@ -315,7 +315,7 @@ pub fn derive_from_str_as_json(input: TokenStream) -> TokenStream {
         impl<#params> std::str::FromStr for #name #generics {
             type Err = serde_json::Error;
 
-            fn from_str(s: &str) -> Result<Self, Self::Err> {
+            fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
```

Comment:
This is just a nit (especially with `std`), but in other macro crates of [mine](https://github.com/jofas/map_macro/blob/f9089319e7e02b2054dbbc9595e9f585ff7f8861/src/_std.rs#L22) I recently started to use the [leading `::`](https://doc.rust-lang.org/reference/paths.html#path-qualifiers) to indicate that the path should be evaluated globally in the prelude and not locally in the module:

```suggestion
            fn from_str(s: &str) -> ::std::result::Result<Self, Self::Err> {
```